code/modules/modular_armor/storage.dm

```diff
@@ -9,44 +9,65 @@
 /obj/item/armor_module/storage
 	icon = 'icons/mob/modular/modular_armor_modules.dmi'
 	icon_state = "mod_is_bag"
-
-	/// Internal storage type
-	var/storage_type = /obj/item/storage/internal/modular
-
-/obj/item/armor_module/storage/can_attach(mob/living/user, obj/item/clothing/suit/modular/parent, silent = FALSE)
+	slot = ATTACHMENT_SLOT_STORAGE
+	///Internal storage of the module. Its parent is switched to the parent item when attached.
+	var/obj/item/storage/internal/storage = /obj/item/storage/internal/modular
+	///If TRUE it will add extra overlays for the items within.
+	var/show_storage = FALSE
+	///Icon for the extra storage overlays.
+	var/show_storage_icon = 'icons/mob/modular/modular_helmet_storage.dmi'
+
+/obj/item/armor_module/storage/Initialize()
 	. = ..()
-	if(!.)
-		return
-	if(parent.installed_storage)
-		if(!silent)
-			to_chat(user,span_warning("There is already an installed storage module."))
-		return FALSE
+	storage = new storage(src)
 
-/obj/item/armor_module/storage/can_detach(mob/living/user, obj/item/clothing/suit/modular/parent, silent = FALSE)
+/obj/item/armor_module/storage/Destroy()
 	. = ..()
-	if(!.)
-		return
-	if(parent.storage && length(parent.storage.contents))
-		if(!silent)
-			to_chat(user, "You can't remove this while there are still items inside")
-		return FALSE
+	QDEL_NULL(storage)
```

Comment:
Is it possible to destroy a module without killing the armor it is on?
Is it possilbe to destroy an armor, but the module on it somewhat survive?

Reply:
If one of the above is true, then it's creating harddels